src/server/gc_worker/applied_lock_collector.rs

```diff
@@ -123,10 +131,21 @@ impl LockObserver {
     }
 
     fn send(&self, locks: Vec<(Key, Lock)>) {
-        match self
+        let res = &mut self
             .sender
-            .schedule(LockCollectorTask::ObservedLocks(locks))
-        {
+            .schedule(LockCollectorTask::ObservedLocks(locks));
+        // Wrapping the fail point in a closure, so we can modify
+        // local variables without return,
+        let mut send_fp = || {
```

Comment:
There's a warning on this line:
```
warning: variable does not need to be mutable
   --> src/server/gc_worker/applied_lock_collector.rs:139:13
    |
139 |         let mut send_fp = || {
    |             ----^^^^^^^
    |             |
    |             help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default
```

However, CI passed?

Reply:
Will be fixed by https://github.com/tikv/tikv/pull/8217/files#diff-dd25bab50f5cfdaeeb301092b7820da3R139.